firebase-database/src/main/java/com/google/firebase/database/FirebaseDatabase.java

```diff
@@ -202,7 +193,9 @@ public DatabaseReference getReferenceFromUrl(@NonNull String url) {
           "Can't pass null for argument 'url' in " + "FirebaseDatabase.getReferenceFromUrl()");
     }
 
-    ParsedUrl parsedUrl = Utilities.parseUrl(url, getEmulatorServiceSettings(this.app));
+    ParsedUrl parsedUrl = Utilities.parseUrl(url);
+    parsedUrl.repoInfo.applyEmulatorSettings(this.emulatorSettings);
```

Comment:
I think this call needs to happen after the if statement below. You probably want to support this:

```
var db = FirebaseDatabase.getInstance('http://foo');
db.useEmulator('localhost');
db.getReferenceFromUrl('http://foo/bar');
````

Please add test as well.

Reply:
There is a test for this case actually.  `getReferenceFromUrl` is overridden by emulator settings.  We'll use the namespace from the URL, but the emulator host/port.

Reply:
What about:

```
var db = FirebaseDatabase.getInstance('http://foo');
db.useEmulator('localhost');
db.getReferenceFromUrl('http://bar/baz');
```

That doesn't throw anymore, does it?

Reply:
Will add that to the test.